src/modules/hmis/hmisUtil.ts

```diff
@@ -612,25 +613,40 @@ export const dataUrlForClientImage = (
   return `data:image/jpeg;base64,${image.base64}`;
 };
 
+// Construct ColumnDefs based on which summary-level CustomDataElements
+// are present on the records.
+//services
 export function getCustomDataElementColumns<
   RowType extends { customDataElements: CustomDataElementFieldsFragment[] }
 >(rows: RowType[]) {
   if (!rows || rows.length === 0) return [];
 
-  // Determine which summary-level CDEs are present on these records.
-  // We can look at the first record because records always resolve all
-  // available CDEs, even if they dont have a value.
-  return rows[0].customDataElements
-    .filter((cde) => cde.displayHooks.includes(DisplayHook.TableSummary))
-    .map((cde) => ({
-      header: cde.label,
-      key: cde.key,
+  function generateColumnDefinition(cded: CustomDataElementFieldsFragment) {
+    return {
+      header: cded.label,
+      key: cded.key,
       render: (row: RowType) => {
         const thisCde = row.customDataElements.find(
```

Comment:
I wonder if it would be possible/desirable to change the api so that `row.customDataElements` returns a map instead of an array?